src/wallet/wallet_tx_builder.cpp

```diff
@@ -501,38 +501,35 @@ std::pair<uint256, uint256> WalletTxBuilder::SelectOVKs(
 
 PrivacyPolicy TransactionEffects::GetRequiredPrivacyPolicy() const
 {
-    PrivacyPolicy maxPrivacy = PrivacyPolicy::FullPrivacy;
-
-    if (!spendable.orchardNoteMetadata.empty() && payments.HasSaplingRecipient()) {
-        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
-    }
-
-    if (!spendable.saplingNoteEntries.empty() && payments.HasOrchardRecipient()) {
-        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
-    }
-
-    if (!spendable.sproutNoteEntries.empty() && payments.HasSaplingRecipient()) {
-        maxPrivacy = PrivacyPolicy::AllowRevealedAmounts;
-    }
-
-    bool hasTransparentSource = !spendable.utxos.empty();
-    if (payments.HasTransparentRecipient()) {
-        if (hasTransparentSource) {
+    if (!spendable.utxos.empty()) {
+        std::set<CScriptID> receivedAddrs;
+        for (auto &utxo : spendable.utxos) {
+            receivedAddrs.emplace(utxo.Script());
```

Comment:
The spend script is not the address; you will need to use https://github.com/zcash/zcash/blob/master/src/script/standard.cpp#L157-L185 to get the address.